servers/src/websocket/farmer/new_proof_or_space.rs

```diff
@@ -60,13 +60,11 @@ impl<T: PoolClient + Sized + Sync + Send + 'static> MessageHandler for NewProofO
         peer_id: Arc<Bytes32>,
         peers: PeerMap,
     ) -> Result<(), Error> {
-        let exists;
-        {
-            exists = peers.lock().await.get(&peer_id).is_some();
-        }
-        if exists {
+        let peer = peers.read().await.get(&peer_id).cloned();
+        if let Some(peer) = peer {
```

Comment:
Building christmas trees is hard to read, maybe we can get rid of all this indentation here and below